src/c_wrappers.jl

```diff
@@ -51,95 +51,81 @@ function create_c_nomad_problem(eval_bb::Function,
-    return is_ok
-end
-
-function set_nomad_display_infeasible!(prob::C_NomadProblem, display_infeasible::Bool)
-    is_ok = ccall((:setNomadDisplayInfeasible, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, display_infeasible)
-    return is_ok
-end
-
-function set_nomad_display_unsuccessful!(prob::C_NomadProblem, display_unsuccessful::Bool)
-    is_ok = ccall((:setNomadDisplayUnsuccessful, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, display_unsuccessful)
-    return is_ok
-end
-
-function set_nomad_opportunistic_eval!(prob::C_NomadProblem, opportunistic_eval::Bool)
-    is_ok = ccall((:setNomadOpportunisticEval, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, opportunistic_eval)
+function add_nomad_param!(prob::C_NomadProblem, keyword_instruction::String)
+    is_ok = ccall((:addNomadParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}), prob.ref, keyword_instruction)
     return is_ok
 end
 
-function set_nomad_use_cache!(prob::C_NomadProblem, use_cache::Bool)
-    is_ok = ccall((:setNomadUseCache, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, use_cache)
+function add_nomad_val_param!(prob::C_NomadProblem, keyword::String, value::Int)
+    is_ok = ccall((:addNomadValParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}, Cint), prob.ref, keyword, value)
     return is_ok
 end
 
-function set_nomad_LH_search_params!(prob::C_NomadProblem, lh_search_init::Int, lh_search_iter::Int)
-    is_ok = ccall((:setNomadLHSearchParams, libnomadInterface), Cint, (Ptr{Cvoid}, Cint, Cint), prob.ref, lh_search_init, lh_search_iter)
+function add_nomad_bool_param!(prob::C_NomadProblem, keyword::String, value::Bool)
+    is_ok = ccall((:addNomadBoolParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}, Cint), prob.ref, keyword, value)
     return is_ok
 end
 
-function set_nomad_speculative_search!(prob::C_NomadProblem, speculative_search::Bool)
-    is_ok = ccall((:setNomadUseCache, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, speculative_search)
+function add_nomad_string_param!(prob::C_NomadProblem, keyword::String, param_str::String)
+    is_ok = ccall((:addNomadStringParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}, Ptr{UInt8}), prob.ref, keyword, param_str)
     return is_ok
 end
 
-function set_nomad_nm_search!(prob::C_NomadProblem, nm_search::Bool)
-    is_ok = ccall((:setNomadUseCache, libnomadInterface), Cint, (Ptr{Cvoid}, Cint), prob.ref, nm_search)
+function add_nomad_array_of_double_param!(prob::C_NomadProblem, keyword::String, a::Vector{Float64})
+    is_ok = ccall((:addNomadArrayOfDoubleParam, libnomadCInterface), Cint, (Ptr{Cvoid}, Ptr{UInt8}, Ptr{Float64}), prob.ref, keyword, a)
     return is_ok
 end
 
-function solve_problem(prob::C_NomadProblem, x0::Vector{Float64})
+function solve_nomad_problem(prob::C_NomadProblem, x0s::Vector{Float64}, nb_starting_points::Int)
```

Comment:
Pourquoi est-ce que tu as changé `x0` en `x0s` ?

Reply:
C style: you can now pass to the C function several starting points, hence the `x0s`. I have not put this feature yet on the core functions.